pygmt/clib/session.py

```diff
@@ -1905,6 +1915,10 @@ def virtualfile_to_grid(
         """
         if outgrid is not None:
             return None
+        if vfname.split("-")[3] == "I":
+            with GMTTempFile(suffix=".tif") as tmpfile:
+                self.call_module("write", f"{vfname} {tmpfile.name} -Ti")
+                return xr.load_dataarray(tmpfile.name)
```

Comment:
A little unsure how this part works, where are you getting the "I" from?

Reply:
The virtual file name has the form of `@GMTAPI@-S-D-F-A-G-M-######`, in which `F` stands for the file family, and `I` means the virtual file is a `GMT_IMAGE`. This is just a quick hacky way to determine the virtual file family. Ideally we should wrap the `GMT_Inquire_VirtualFile` API function (http://docs.generic-mapping-tools.org/dev/devdocs/api.html#gmt-inquire-virtualfile).

See https://github.com/GenericMappingTools/gmt/blob/e5ef7016eaccce3fac58419f81249cc59d8735d4/src/gmt_api.c#L3364 for details.